packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1709,31 +1708,17 @@ class DropdownButtonFormField<T> extends FormField<T> {
               skipTraversal: true,
               child: Builder(builder: (BuildContext context) {
                 final bool isFocused = Focus.of(context).hasFocus;
-                InputBorder? resolveInputBorder() {
-                  if (hasError) {
-                    if (isFocused) {
-                      return effectiveDecoration.focusedErrorBorder;
-                    }
-                    return effectiveDecoration.errorBorder;
-                  }
-                  if (isFocused) {
-                    return effectiveDecoration.focusedBorder;
-                  }
-                  if (effectiveDecoration.enabled) {
-                    return effectiveDecoration.enabledBorder;
-                  }
-                  return effectiveDecoration.border;
-                }
-                BorderRadius? effectiveBorderRadius() {
-                  final InputBorder? inputBorder = resolveInputBorder();
-                  if (inputBorder is OutlineInputBorder) {
-                    return inputBorder.borderRadius;
-                  }
-                  if (inputBorder is UnderlineInputBorder) {
-                    return inputBorder.borderRadius;
-                  }
-                  return null;
-                }
+                late final InputBorder? resolvedBorder = switch ((
+                  enabled: effectiveDecoration.enabled,
+                  focused: isFocused,
+                  error: effectiveDecoration.errorText != null,
+                )) {
+                  (enabled: _, focused: true, error: true) => effectiveDecoration.focusedErrorBorder,
```

Comment:
Anonymous parameter names are in the style guide under things to avoid. Can we go without them here?
https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-anonymous-parameter-names

Reply:
The underscore here is not a parameter name though, it's a pattern that matches anything.